packages/core/docs/composition/composables.md

```diff
@@ -1,128 +1,108 @@
+3. Finally, we return the `user` object from the `setup` method to make it available in the components `<template>`.
 
-## What composables I can use
+While it's okay to destructure composable like we did in step 1, you should **not** destructure read-only state, such as the `user` or `error` properties, because resulting variables will not be reactive and won't update.
 
-Vue Storefront integrations are exposing the following composables:
+```javascript
+// ❌ Destructuring `user` will create variables that aren't reactivity and don't update
+const { user: { value: { firstname } } } = useUser();
 
-#### Product Catalog
-
-- [`useProduct`](../reference/api/core.useproduct.html) - Managing a single product with variants (or a list).
-- [`useCategory`](../reference/api/core.usecategory.html) - Managing category lists (but not category products).
-- [`useFacet`](../reference/api/core.usefacet.html) - Complex catalog search with filtering.
-- [`useReview`](../reference/api/core.usereview.html) - Product reviews.
-
-#### User Profile and Authorization
-
-- [`useUser`](../reference/api/core.useuser.html) - Managing user sessions, credentials and registration.
-- [`useUserShipping`](../reference/api/core.useusershipping.html) - Managing shipping addresses.
-- [`useUserBilling`](../reference/api/core.useuserbilling.html) - Managing billing addresses.
-- [`useUserOrder`](../reference/api/core.useuserorder.html) - Managing past and active user orders.
+// ✔️ Using `computed` will make the variable react to changes in the `user` object
+const { user } = useUser();
+const firstname = computed(() => user.value.firstname);
+```
 
-#### Shopping Cart
+This raises two questions:
 
-- [`useCart`](../reference/api/core.usecart.html) - Loading the cart, adding/removing products and discounts.
+1. What is the `useFetch`, and what does it do?
+2. What happened when we called the `load` method?
 
-#### Wishlist/Favourite
+### `useFetch` and other hooks for fetching data
 
-- [`useWishlist`](../reference/api/core.usewishlist.html) - Loading the wishlist, adding/removing products.
+There are many hooks available in Composition API, but let's only focus on the most common ones used for fetching data:
 
-#### CMS Content
+- The [useFetch](https://composition-api.nuxtjs.org/lifecycle/usefetch/) and [useAsync](https://composition-api.nuxtjs.org/API/useAsync) are Nuxt-specific hooks called on the server-side when rendering the route and on the client-side when navigating between pages.
+- The `onMounted` is a lifecycle hook called **only on the client-side** after the browser loads the page.
+- The `onServerPrefetch` is a lifecycle hook called **only on the server-side** when rendering the route.
 
-- [`useContent`](../reference/api/core.usecontent.html) - Fetching the CMS content. It is usually used in combination with `<RenderContent>`component.
+You can use one or more hooks simultaneously, even in the same component.
 
-#### Checkout
+### Internals of the `load` method
 
```

Comment:
"to reflect the current state, and later the result of the API call corresponding to this method." -> "to reflect the current state, makes an API call, and then updates the state with the API's response."